src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -60,6 +60,13 @@ public IEnumerable<string> Scopes
         /// </summary>
         public string Authority { get; set; }
 
+        /// <summary>
+        /// AAD tenant ID to which the Azure subscription belongs to.
+        /// </summary>
+        public string TenantId => Instance.Authority
+                                          .CreateAuthority(Authority, validateAuthority: true)
+                                          .TenantId;
```

Comment:
Also to @jmprieur 's point then, let's remove this helper method, as it is tied to an MSAL implementation:

```csharp
 public static Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(IMsalHttpClientFactory httpClientFactory, string resourceUri, CancellationToken cancellationToken=default(CancellationToken));
```